Can't load language parsers in combination with Neovim

<!-- 
Thanks for letting us know about a bug! 💖

Use the questions below to help explain what you think is wrong, and what we should do to fix it.
-->

### 👓 What did you see?

Hello. I am trying to use `cucumber_language_server` in combination with NeoVim (specifically LunarVim). Here is the related LSP configuration:
```lua
require("lvim.lsp.manager").setup("cucumber_language_server", {
  settings = {
    cucumber = {
      features = { 'features/**/*.feature' },
      glue = { 'features/step_definitions/*.rb', 'features/step_definitions/*.ts' }
    }
  }
})
```

This configuration is correctly passed to `cucumber_language_server`. The features & glue are correctly loaded (I can see this in the logs).

The problem is that the step definitions are not able to be parsed. There are many errors related to this in my LSP logs. I have attached my logs in this bug report, see the bottom of this issue. I think the following line tells us that the Ruby parser (or any other language parser) can not be loaded:
```
ENOENT: no such file or directory, open 'file:///Users/pawel/.local/share/nvim/mason/packages/cucumber-language-server/node_modules/@cucumber/language-server/node_modules/@cucumber/language-service/dist/ruby.wasm'
```
It is very strange that this file can not be loaded, because I have manually confirmed that the `ruby.wasm` file exists in exactly that directory.

At the bottom of the log file you can see that there are errors when trying to parse the Ruby file containing the step definition:
```
[ERROR][2022-11-09 19:34:43] ...lsp/handlers.lua:484    "* Step Definition errors: Error: Parsing failed
uri: file:///Users/pawel/Projects/cucumber_test/features/step_definitions/test.rb
language: ruby

    at Parser.parse (/Users/pawel/.local/share/nvim/mason/packages/cucumber-language-server/node_modules/@cucumber/language-server/node_modules/web-tree-sitter/tree-sitter.js:1:38190)
    at SourceAnalyzer.parse (/Users/pawel/.local/share/nvim/mason/packages/cucumber-language-server/node_modules/@cucumber/language-server/node_modules/@cucumber/language-service/src/language/SourceAnalyzer.ts:139:72)
    at SourceAnalyzer.getSourceMatches (/Users/pawel/.local/share/nvim/mason/packages/cucumber-language-server/node_modules/@cucumber/language-server/node_modules/@cucumber/language-service/src/language/SourceAnalyzer.ts:107:21)
    at SourceAnalyzer.eachParameterTypeLink (/Users/pawel/.local/share/nvim/mason/packages/cucumber-language-server/node_modules/@cucumber/language-server/node_modules/@cucumber/language-service/src/language/SourceAnalyzer.ts:35:39)
    at ExpressionBuilder.build (/Users/pawel/.local/share/nvim/mason/packages/cucumber-language-server/node_modules/@cucumber/language-server/node_modules/@cucumber/language-service/src/language/ExpressionBuilder.ts:45:20)
    at CucumberLanguageServer.<anonymous> (/Users/pawel/.local/share/nvim/mason/packages/cucumber-language-server/node_modules/@cucumber/language-server/src/CucumberLanguageServer.ts:417:59)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/pawel/.local/share/nvim/mason/packages/cucumber-language-server/node_modules/@cucumber/language-server/dist/cjs/src/CucumberLanguageServer.js:5:58)"
```
My intuition tells me that the step definition can not be parsed because the Ruby parser is not loaded somehow.

I have opened the same folder in VSCode, and the LSP works fine there. The logs are a bit different from my Neovim LSP logs, though.

### ✅ What did you expect to see?

The step definition file is correctly parsed, and I get full LSP support in Neovim

### 📦 Which tool/library version are you using?

`cucumber_language_server` 1.1.1

### 🔬 How could we reproduce it?

The easiest way to reproduce it would be to install LunarVim. Note: This error is not unique to LunarVim. LunarVim is a custom NeoVim distro that uses `nvim-lsp` to connect to language servers. This problem will probably occur on any NeoVim distro.

1. Create the following file structure:
```
repro_example
|- features
          |- step_definitions
          |              |- test.rb
          |- test.feature
```
2. `test.feature` content:
```cucumber
Feature: Test
  Scenario: A test scenario
    Given a step definition
```
3. `test.rb` content:
```ruby
Given('a step definition') do
  puts 'Hello World'
end
```
4. [Install LunarVim](https://www.lunarvim.org/docs/installation#release)
5. Edit `~/.config/lvim/config.lua` and add the following LSP configuration:
```lua
require("lvim.lsp.manager").setup("cucumber_language_server", {
  settings = {
    cucumber = {
      features = { 'features/**/*.feature' },
      glue = { 'features/step_definitions/*.rb', 'features/step_definitions/*.ts' }
    }
  }
})
```
7. Use `lvim` to open LunarVim
8. Type `:LspInstall cucumber` and select `cucumber_language_server`
9. Wait for it to install, then quit LunarVim
10. Open `test.feature` in LunarVim in your shell using `lvim test.feature`
11. Assert that the steps are undefined and they don't work.

Logs can be found in `~/local/state/nvim/lsp.log`. 

### 📚 Any additional context?

<!-- Add any other context, references or screenshots about the problem here. -->
[lsp.log](https://github.com/cucumber/language-server/files/9974096/lsp.log)

----
 
*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/ISSUE_TEMPLATE/bug_report.md)*

hello, I am facing the same issue when installing the cucumber-ls with Mason in Neovim. I found a branch 'fix-standalone', which seems to work, but stops working if I update it with master.

`[ERROR][2022-12-16 10:19:37] .../vim/lsp/rpc.lua:733	"rpc"	"cucumber-language-server"	"stderr"	"Error: Cannot find module '../src/version'\nRequire stack:\n- /home/laurens/.local/share/nvim/mason/packages/cucumber-language-server/node_modules/@cucumber/language-server/bin/cucumber-language-server.cjs\n    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1039:15)\n    at Function.Module._load (node:internal/modules/cjs/loader:885:27)\n    at Module.require (node:internal/modules/cjs/loader:1105:19)\n    at require (node:internal/modules/cjs/helpers:103:18)\n    at Object.<anonymous> (/home/laurens/.local/share/nvim/mason/packages/cucumber-language-server/node_modules/@cucumber/language-server/bin/cucumber-language-server.cjs:7:21)\n    at Module._compile (node:internal/modules/cjs/loader:1218:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1272:10)\n    at Module.load (node:internal/modules/cjs/loader:1081:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:922:12)\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:82:12)\n"
`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 ### Fixed
 - Fixed c-sharp glob paths for step definitions and feature files - [#89](https://github.com/cucumber/language-server/pull/89)
 - Specify minimum supported node version ([#100](https://github.com/cucumber/language-server/pull/100))
+- Fixed the issue preventing standalone operation outside of VS Code - [#74](https://github.com/cucumber/language-server/pull/74)
 
 ### Added
 - Allow Javascript/Typescript glue files with the following file extensions: cjs, mjs, cts, mts - [#85](https://github.com/cucumber/language-server/pull/85)

--- a/README.md
+++ b/README.md
@@ -22,3 +22,9 @@ provide them.
 The server retrieves `cucumber.*` settings from the client with a [workspace/configuration](https://microsoft.github.io/language-server-protocol/specification#workspace_configuration) request.
 
 See [Settings](https://github.com/cucumber/language-server/blob/main/src/types.ts) for details about the expected format.
+
+## External VSCode Usage
+
+We've encountered an issue with the Node version used by [Treesitter](https://github.com/tree-sitter/tree-sitter/issues/2338), a
+dependency of this language server, when working outside of VSCode. For optimal
+compatibility, please use the same Node version as version 18 of VSCode.

--- a/bin/cucumber-language-server.cjs
+++ b/bin/cucumber-language-server.cjs
@@ -3,13 +3,11 @@
 require('source-map-support').install()
 const { startStandaloneServer } = require('../dist/cjs/src/wasm/startStandaloneServer')
 const { NodeFiles } = require('../dist/cjs/src/node/NodeFiles')
-const url = require('url')
-const { version } = require('../src/version')
+const path = require('path')
+const { version } = require('../dist/cjs/src/version')
 
-const wasmBaseUrl = url.pathToFileURL(
-  `${__dirname}/../node_modules/@cucumber/language-service/dist`
-)
-const { connection } = startStandaloneServer(wasmBaseUrl.href, (rootUri) => new NodeFiles(rootUri))
+const wasmBasePath = path.resolve(`${__dirname}/../node_modules/@cucumber/language-service/dist`)
+const { connection } = startStandaloneServer(wasmBasePath, (rootUri) => new NodeFiles(rootUri))
 
 // Don't die on unhandled Promise rejections
 process.on('unhandledRejection', (reason, p) => {

--- a/src/wasm/startStandaloneServer.ts
+++ b/src/wasm/startStandaloneServer.ts
@@ -12,4 +12,8 @@ export function startStandaloneServer(wasmBaseUrl: string, makeFiles: (rootUri: 
   const documents = new TextDocuments(TextDocument)
   new CucumberLanguageServer(connection, documents, adapter, makeFiles, () => undefined)
   connection.listen()
+
+  return {
+    connection,
+  }
 }

--- a/test/CucumberLanguageServer.test.ts
+++ b/test/CucumberLanguageServer.test.ts
@@ -1,7 +1,7 @@
 import { WasmParserAdapter } from '@cucumber/language-service/wasm'
 import assert from 'assert'
 import { Duplex } from 'stream'
-import { Logger, StreamMessageReader, StreamMessageWriter } from 'vscode-jsonrpc/node'
+import { NullLogger, StreamMessageReader, StreamMessageWriter } from 'vscode-jsonrpc/node'
 import {
   Connection,
   createProtocolConnection,
@@ -80,11 +80,10 @@ describe('CucumberLanguageServer', () => {
       },
       workspaceFolders: null,
     }
-    const logger = new NullLogger()
     clientConnection = createProtocolConnection(
       new StreamMessageReader(outputStream),
       new StreamMessageWriter(inputStream),
-      logger
+      NullLogger
     )
     clientConnection.onError((err) => {
       console.error('ERROR', err)
@@ -203,24 +202,6 @@ class TestStream extends Duplex {
   }
 
   _read() {
-    // no-op
-  }
-}
-
-class NullLogger implements Logger {
-  error(): void {
-    // no-op
-  }
-
-  warn(): void {
-    // no-op
-  }
-
-  info(): void {
-    // no-op
-  }
-
-  log(): void {
     // no-op
   }
 }

--- a/test/standalone.test.ts
+++ b/test/standalone.test.ts
@@ -1,0 +1,125 @@
+import assert from 'assert'
+import { ChildProcess, fork } from 'child_process'
+import { Duplex } from 'stream'
+import { NullLogger, StreamMessageReader, StreamMessageWriter } from 'vscode-jsonrpc/node'
+import {
+  createProtocolConnection,
+  DidChangeConfigurationNotification,
+  DidChangeConfigurationParams,
+  InitializeParams,
+  InitializeRequest,
+  LogMessageNotification,
+  LogMessageParams,
+  ProtocolConnection,
+} from 'vscode-languageserver'
+
+import { Settings } from '../src/types.js'
+
+describe('Standalone', () => {
+  let serverFork: ChildProcess
+  let logMessages: LogMessageParams[]
+  let clientConnection: ProtocolConnection
+
+  beforeEach(async () => {
+    logMessages = []
+    serverFork = fork('./bin/cucumber-language-server.cjs', ['--stdio'], {
+      silent: true,
+    })
+
+    const initializeParams: InitializeParams = {
+      rootUri: `file://${process.cwd()}`,
+      processId: NaN, // This id is used by vscode-languageserver. Set as NaN so that the watchdog responsible for watching this process does not run.
+      capabilities: {
+        workspace: {
+          configuration: true,
+          didChangeWatchedFiles: {
+            dynamicRegistration: true,
+          },
+        },
+        textDocument: {
+          moniker: {
+            dynamicRegistration: false,
+          },
+          completion: {
+            completionItem: {
+              snippetSupport: true,
+            },
+          },
+          semanticTokens: {
+            tokenTypes: [],
+            tokenModifiers: [],
+            formats: [],
+            requests: {},
+          },
+          formatting: {
+            dynamicRegistration: true,
+          },
+        },
+      },
+      workspaceFolders: null,
+    }
+    if (!serverFork.stdin || !serverFork.stdout) {
+      throw 'Process created without stdio streams'
+    }
+    clientConnection = createProtocolConnection(
+      new StreamMessageReader(serverFork.stdout as Duplex),
+      new StreamMessageWriter(serverFork.stdin as Duplex),
+      NullLogger
+    )
+    clientConnection.onError((err) => {
+      console.error('ERROR', err)
+    })
+    clientConnection.onNotification(LogMessageNotification.type, (params) => {
+      if (params.type !== 3) {
+        logMessages.push(params)
+      }
+    })
+    clientConnection.onUnhandledNotification((n) => {
+      console.error('Unhandled notification', n)
+    })
+    clientConnection.listen()
+    const { serverInfo } = await clientConnection.sendRequest(
+      InitializeRequest.type,
+      initializeParams
+    )
+    assert.strictEqual(serverInfo?.name, 'Cucumber Language Server')
+  })
+
+  afterEach(() => {
+    clientConnection.end()
+    clientConnection.dispose()
+    serverFork.kill('SIGTERM') // Try to terminate first
+    serverFork.kill('SIGKILL') // Then try to kill if it is not killed
+  })
+
+  context('workspace/didChangeConfiguration', () => {
+    it(`startup success`, async () => {
+      // First we need to configure the server, telling it where to find Gherkin documents and Glue code.
+      // Note that *pushing* settings from the client to the server is deprecated in the LSP. We're only using it
+      // here because it's easier to implement in the test.
+      const settings: Settings = {
+        features: ['testdata/**/*.feature'],
+        glue: ['testdata/**/*.js'],
+        parameterTypes: [],
+        snippetTemplates: {},
+      }
+      const configParams: DidChangeConfigurationParams = {
+        settings,
+      }
+
+      await clientConnection.sendNotification(DidChangeConfigurationNotification.type, configParams)
+
+      await new Promise((resolve) => setTimeout(resolve, 1000))
+
+      assert.strictEqual(
+        logMessages.length,
+        // TODO: change this to 0 when `workspace/semanticTokens/refresh` issue was solved
+        1,
+        // print readable log messages
+        logMessages
+          .map(({ type, message }) => `**Type**: ${type}\n**Message**:\n${message}\n`)
+          .join('\n--------------------\n')
+      )
+    })
+  })
+})

